src/test/unittest/ut.c

```diff
@@ -593,3 +607,25 @@ ut_checksum(uint8_t *addr, size_t len)
 
 	return (uint16_t)(sum2 << 8) | sum1;
 }
+
+/*
+ * ut_append_array -- append new element to array
+ * returns increased array and new size by argument
+ */
+char **
+ut_append_to_array(int size, const char *array[],
```

Comment:
Seems to me you want to grow a buffer and append an element to it.
This is not the way to do it:
1. You might be leaking memory.
2. Without looking at the code, the programmer might think that the array pointer will be reallocated.
3. You can achieve the same thing with a simple realloc(3)
4. No error handling.
5. It's almost never a good idea to allocate memory in one compilation unit, and require the user to free it in another one.